scripts/update_plugin.sh

```diff
@@ -16,36 +16,78 @@ cancel() {
 	exit 0
 }
 
-pull_changes() {
-	local plugin="$1"
-	local plugin_path=$(shared_plugin_path "$plugin")
-	cd $plugin_path &&
-		GIT_TERMINAL_PROMPT=0 git pull &&
-		GIT_TERMINAL_PROMPT=0 git submodule update --init --recursive
+update_plugin_git() {
+	local plugin; plugin="$1"
+	local branch; branch=":${1##*:}"
+	local plugin_path; plugin_path=$(shared_plugin_path "$plugin")
+	plugin="${1%$branch}"
+	#update only makes sense when no specific branch/revision is set
+	if [ "${branch}" = ":${1}" ]; then
+		 cd "$plugin_path" &&
+			  GIT_TERMINAL_PROMPT="0" git pull &&
+			  GIT_TERMINAL_PROMPT="0" git submodule update --init --recursive
```

Comment:
unrelated to this specific pull request, but couldn't these two be combined in a single `git pull --recursive` nowadays?